Tools/CMake/AMReXTargetHelpers.cmake

```
@@ -107,3 +107,17 @@ function (set_cpp_sources_to_cuda_language _target)
    list(FILTER _sources INCLUDE REGEX "\\.cpp")
    set_source_files_properties(${_sources} PROPERTIES LANGUAGE CUDA )
 endfunction ()
+
+#
+# Setup an amrex-dependent target for cuda compilation.
+# This function ensures that the CUDA compilation of _target
+# is compatible with amrex CUDA build.
+#
+function (setup_target_for_cuda_compilation _target)
+   set_target_properties( ${_target}
```

I added the following comment in WarpX to recall why we have to set this at all:
```suggestion
   # mainly due to amrex::Random which uses global device variables
   set_target_properties( ${_target}
```
https://github.com/ECP-WarpX/WarpX/pull/759